SpotifyArtistsViewer/SARequestManager.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  SARequestManager.swift
+//  SpotifyArtistsViewer
+//
+//  Created by Kevin Zeckser on 6/6/16.
+//  Copyright © 2016 Kevin Zeckser. All rights reserved.
+//
+import Foundation
+
+enum SAResponse {
+    case Success(artists: [SAArtist])
+    case Failure(error: ErrorType)
+}
+
+
+final class SARequestManager {
+    
+    // MARK: - Properties
+    
+    static let sharedInstance = SARequestManager()
+    private let session: NSURLSession!
+    // This prevents others from using the default '()' initializer for this class.
+    private init() {
+        let sessionConfig: NSURLSessionConfiguration = NSURLSessionConfiguration.defaultSessionConfiguration()
+        session = NSURLSession(configuration: sessionConfig)
+    }
+    
+    // MARK: - NSURLRequest & JSON Parse
+    
+    func getArtistsWithQuery(query: String, completion: SAResponse -> Void) {
+        let path = "https://api.spotify.com/v1/search?q=\(query)&type=track,artist&market=US"
+        guard let url = NSURL(string: path) else { return }
```

Comment:
you can also have a property for the NSURL, which has a getter that does this check for you.